torchvision/datasets/video_utils.py

```diff
@@ -0,0 +1,153 @@
+import bisect
+import torch
+import torch.utils.data
+from torchvision.io import read_video_timestamps, read_video
+
+
+def unfold(tensor, size, step, dilation):
+    """
+    similar to tensor.unfold, but with the dilation
+    and specialized for 1d tensors
+
+    Returns all consecutive windows of `size` elements, with
+    `step` between windows. The distance between each element
+    in a window is given by `dilation`.
+    """
+    assert tensor.dim() == 1
+    o_stride = tensor.stride(0)
+    numel = tensor.numel()
+    new_stride = (step * o_stride, dilation * o_stride)
+    new_size = ((numel - (dilation * (size - 1) + 1)) // step + 1, size)
+    if new_size[0] < 1:
+        new_size = (0, size)
+    return torch.as_strided(tensor, new_size, new_stride)
+
+
+class VideoClips(object):
+    """
+    Given a list of video files, computes all consecutive subvideos of size
+    `clip_length_in_frames`, where the distance between each subvideo in the
+    same video is defined by `frames_between_clips`.
+
+    Creating this instance the first time is time-consuming, as it needs to
+    decode all the videos in `video_paths`. It is recommended that you
```

Comment:
Since you mentioned creating this object is time-consuming, do you want to implement methods such as `save` and `load` to support caching ?

Reply:
I think @fmassa  was hoping to keep that separate from the torchvision per se, and keep it in examples? 

Regardless where we put though it this is a great idea!

Reply:
I was thinking that this is something that can be achieved by just saving the `VideoClips` object, and showcase how to do it in the `references/video_classification` example that I'm planning to add.

So the user would just need to do
```python
torch.save(video_clips, 'cache.pth')
...
video_clips = torch.load('cache.pth')
```

Thoughts?

Reply:
I agree with that

Reply:
@fmassa that sounds good.